### PR TITLE
ISVAPP-1401: health check added before supplying hosts to connection pool

### PR DESCRIPTION
**Pull request by sarmuru2**
When one of the gemini nodes are down, galaxy is not starting. Instead change it to start galaxy when n/2+1 nodes are up.

### CODE REVIEW

**Review by manjurad — APPROVED** [review]

File: redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/ConfigurationHostSupplier.java [hunk 1]
```diff
@@ -34,9 +37,80 @@ public ConfigurationHostSupplier(RedisProperties properties) {
...
+                    "Successful dynomite hosts size is less than majority. Hence conductor will not connect to redis");
+        }
+    }
+
+    private boolean ping(Host host) {
+        JedisPool jedisPool = null;
+        Jedis jedis = null;
+        try {
+            JedisPoolConfig config = new JedisPoolConfig();
+            config.setMinIdle(1);
```
Comment (manjurad) [inline redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/ConfigurationHostSupplier.java:82]: is this setting the minimum and maximum number of connections? Can you add comments for what this setting does

File: redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/ConfigurationHostSupplier.java [hunk 2]
```diff
@@ -34,9 +37,80 @@ public ConfigurationHostSupplier(RedisProperties properties) {
...
+            config.setMaxTotal(1);
+            if (host.getPassword() == null) {
+                jedisPool = new JedisPool(config, host.getHostName(), host.getPort());
```
Comment (manjurad) [inline redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/ConfigurationHostSupplier.java:85]: in this case what is the timeout value? I guess in the second case the timeout is 3000 ms? Can you make it consistent

File: redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/ConfigurationHostSupplier.java (hunk 1 above)
Comment (sarmuru2) [reply redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/ConfigurationHostSupplier.java:98]: its minimum number of connections that can be idle. and the maximum connections that can be created for this instance.

File: redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/ConfigurationHostSupplier.java (hunk 2 above)
Comment (sarmuru2) [reply redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/ConfigurationHostSupplier.java]: fixed

**Pull request merged**